src/cms/views/language_tree/language_tree_actions.py

```diff
@@ -67,3 +68,43 @@ def move_language_tree_node(
         logger.exception(e)
 
     return redirect("language_tree", **{"region_slug": region_slug})
+
+
+def delete_language_tree_node(request, region_slug, language_tree_node_id):
+    """
+    Deletes the language node of distinct region
+    and all page translations for this language
+
+    :param request: The current request
+    :type request: ~django.http.HttpResponse
+
+    :param region_slug: The slug of the region which language node should be deleted
+    :type region_slug: str
+
+    :param language_tree_node_id: The id of the language tree node wich should be deleted
+    :type language_tree_node_id: int
+
+    :return: A redirection to the language tree
+    :rtype: ~django.http.HttpResponseRedirect
+    """
+    try:
+        # get current selected language node
+        language_node = LanguageTreeNode.objects.get(pk=language_tree_node_id)
+        # get all page translation assigned to the language node
+        page_translations = language_node.language.page_translations
+        # filter these translation that belong to the region
+        current_region_translations = page_translations.filter(
+            page__region__slug=region_slug
+        )
+        current_region_translations.delete()
+        language_node.delete()
+        messages.success(
+            request,
+            _(
+                'The language tree node "{}" and all corresponding page translations were successfully deleted.'
+            ).format(language_node.translated_name),
+        )
+    except (ObjectDoesNotExist) as e:
+        messages.error(request, e)
+        logger.exception(e)
```

Comment:
```suggestion
    # get current region
    region = Region.get_current_region(request)
    # get current selected language node
    language_node = get_object_or_404(region.language_tree_nodes, id=language_tree_node_id)
    # get all page translation assigned to the language node
    page_translations = language_node.language.page_translations
    # filter these translation that belong to the region
    current_region_translations = page_translations.filter(
        page__region=region
    )
    current_region_translations.delete()
    language_node.delete()
    messages.success(
        request,
        _(
            'The language tree node "{}" and all corresponding page translations were successfully deleted.'
        ).format(language_node.translated_name),
    )
```
I propose the following changes:
- Use `get_object_or_404()` instead of try-catch-block
- Only delete a language tree node if it belongs to the current region (otherwise, region users could delete other region's language tree ndoes)